stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/EventBuilderImpl.java

```diff
@@ -125,12 +128,30 @@ public EntityDeleteResults queueEntityDelete( final ApplicationScope application
 
         final GraphManager gm = graphManagerFactory.createEdgeManager( applicationScope );
 
+        //needs get versions here.
+
+
+        //TODO: change this to be an observable
+        //so we get these versions and loop through them until we find the MvccLogEntry that is marked as delete.
+        //TODO: evauluate this to possibly be an observable to pass to the nextmethod.
+        MvccLogEntry mostRecentlyMarked = ecm.getVersions( entityId ).toBlocking()
+                                             .firstOrDefault( null,
```

Comment:
Should be combined with the observable below. There isn't any reasons we should go through entity versions twice.